File: Capital/下單API_2.1.22/DotNet範例(VS2010)/OrderTesterVC/ForeignStockOrderControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace OrderTester
{
    public partial class ForeignStockOrderControl : UserControl
    {
        #region Define Variable
        //----------------------------------------------------------------------
        // Define Variable
        //----------------------------------------------------------------------

        private int m_nCode;
        public string m_strMessage;

        public delegate void MyMessageHandler(string strType, int nCode, string strMessage);
        public event MyMessageHandler GetMessage;

        private string m_UserAccount = "";
        public string UserAccount
        {
            get { return m_UserAccount; }
            set { m_UserAccount = value; }
        }

        #endregion

        #region Initialize
        //----------------------------------------------------------------------
        // Initialize
        //----------------------------------------------------------------------
        public ForeignStockOrderControl()
        {
            InitializeComponent();
        }
        #endregion

        #region Component Event
        //----------------------------------------------------------------------
        // Component Event
        //----------------------------------------------------------------------
        private void btnSendForeignStockOrder_Click(object sender, EventArgs e)
        {
            if (m_UserAccount == "")
            {
                MessageBox.Show("請選擇複委託帳號");
                return;
            }
            
            string      strExchangeNo = "";
            string      strStockNo;
            int         nBidAsk;
            string      strPrice;
            int         nQty;
            string      strCurrency1;
            string      strCurrency2;
            string      strCurrency3;
            int         nAccountType;


            if (boxAccountType.SelectedIndex < 0)
            {
                MessageBox.Show("請選擇帳戶別");
                return;
            }
            nAccountType = boxAccountType.SelectedIndex + 1;

            if (boxCurrency1.SelectedIndex < 0)
            {
                MessageBox.Show("請至少選擇扣款幣別 1");
                return;
            }
            strCurrency1 = boxCurrency1.Text;
            strCurrency2 = boxCurrency2.Text;
            strCurrency3 = boxCurrency3.Text;

            if (boxExchange.SelectedIndex < 0)
            {
                MessageBox.Show("請選擇交易所");
                return;
            }

            if (boxExchange.SelectedIndex == 0)
            {
                strExchangeNo = "US";
            }
            else if (boxExchange.SelectedIndex == 1)
            {
                strExchangeNo = "HK";
            }

            if (txtStockNo.Text.Trim() == "")
            {
                MessageBox.Show("請輸入商品代碼");
                return;
            }
            strStockNo = txtStockNo.Text.Trim();

            if (boxBidAsk.SelectedIndex < 0)
            {
                MessageBox.Show("請選擇買賣別");
                return;
            }
            nBidAsk = boxBidAsk.SelectedIndex;

            double dPrice = 0.0;
            if (double.TryParse(txtPrice.Text.Trim(), out dPrice) == false)
            {
                MessageBox.Show("委託價請輸入數字");
                return;
            }
            strPrice = txtPrice.Text.Trim();

            if (int.TryParse(txtQty.Text.Trim(), out nQty) == false)
            {
                MessageBox.Show("委託量請輸入數字");
                return;
            }

            StringBuilder strMessage = new StringBuilder();
            strMessage.Length = 1024;
            Int32 nSiz = 1024;

            m_nCode = Functions.SendForeignStockOrder(m_UserAccount, strStockNo, strExchangeNo, nBidAsk, strPrice, nQty, strCurrency1, strCurrency2, strCurrency3, nAccountType, strMessage, out nSiz);

            m_strMessage = strMessage.ToString();

            if (GetMessage != null)
            {
                GetMessage("ForeignStockOrder", m_nCode, m_strMessage);
            }

            strMessage = null;

        }

        private void btnSendForeignStockOrderAsync_Click(object sender, EventArgs e)
        {
            if (m_UserAccount == "")
            {
                MessageBox.Show("請選擇複委託帳號");
                return;
            }

            string strExchangeNo = "";
            string strStockNo;
            int nBidAsk;
            string strPrice;
            int nQty;
            string strCurrency1;
            string strCurrency2;
            string strCurrency3;
            int nAccountType;


            if (boxAccountType.SelectedIndex < 0)
            {
                MessageBox.Show("請選擇帳戶別");
                return;
            }
            nAccountType = boxAccountType.SelectedIndex + 1;

            if (boxCurrency1.SelectedIndex < 0)
            {
                MessageBox.Show("請至少選擇扣款幣別 1");
                return;
            }
            strCurrency1 = boxCurrency1.Text;
            strCurrency2 = boxCurrency2.Text;
            strCurrency3 = boxCurrency3.Text;

            if (boxExchange.SelectedIndex < 0)
            {
                MessageBox.Show("請選擇交易所");
                return;
            }

            if (boxExchange.SelectedIndex == 0)
            {
                strExchangeNo = "US";
            }
            else if (boxExchange.SelectedIndex == 1)
            {
                strExchangeNo = "HK";
            }

            if (txtStockNo.Text.Trim() == "")
            {
                MessageBox.Show("請輸入商品代碼");
                return;
            }
            strStockNo = txtStockNo.Text.Trim();

            if (boxBidAsk.SelectedIndex < 0)
            {
                MessageBox.Show("請選擇買賣別");
                return;
            }
            nBidAsk = boxBidAsk.SelectedIndex;

            double dPrice = 0.0;
            if (double.TryParse(txtPrice.Text.Trim(), out dPrice) == false)
            {
                MessageBox.Show("委託價請輸入數字");
                return;
            }
            strPrice = txtPrice.Text.Trim();

            if (int.TryParse(txtQty.Text.Trim(), out nQty) == false)
            {
                MessageBox.Show("委託量請輸入數字");
                return;
            }
            StringBuilder strMessage = new StringBuilder();
            strMessage.Length = 1024;
            Int32 nSiz = 1024;

            m_nCode = Functions.SendForeignStockOrderAsync(m_UserAccount, strStockNo, strExchangeNo, nBidAsk, strPrice, nQty, strCurrency1, strCurrency2, strCurrency3, nAccountType, strMessage, out nSiz);

            if (GetMessage != null)
            {
                GetMessage("ForeignStockOrderAsync", m_nCode, "非同步委託");
            }
        }

        private void btnCancelOrder_Click(object sender, EventArgs e)
        {
            string strExchangeNo    = "";
            string strBookNo        = "";

            if (m_UserAccount == "")
            {
                MessageBox.Show("請選擇複委託帳號");
                return;
            }
             
            if (boxExchange.SelectedIndex < 0)
            {
                MessageBox.Show("請選擇交易所");
                return;
            }

            if (boxExchange.SelectedIndex == 0)
            {
                strExchangeNo = "US";
            }
            else if (boxExchange.SelectedIndex == 1)
            {
                strExchangeNo = "HK";
            }

            if (txtCancelBookNo.Text.Trim() == "")
            {
                MessageBox.Show("請輸入欲刪除的委託書號");
                return;
            }
            strBookNo = txtCancelBookNo.Text.Trim();


            StringBuilder strMessage = new StringBuilder();
            strMessage.Length = 1024;
            Int32 nSiz = 1024;

            m_nCode = Functions.CancelForeignStockOrderByBookNo(m_UserAccount,strBookNo,strExchangeNo, strMessage, out nSiz);

            m_strMessage = strMessage.ToString();

            if (GetMessage != null)
            {
                GetMessage("CancelForeignStockOrder By BookNo", m_nCode, m_strMessage);
            }

            strMessage = null;

        }

        private void btnCancelOrderBySeqNo_Click(object sender, EventArgs e)
        {
            string strExchangeNo = "";
            string strSeqNo     = "";

            if (m_UserAccount == "")
            {
                MessageBox.Show("請選擇複委託帳號");
                return;
            }

            if (boxExchange.SelectedIndex < 0)
            {
                MessageBox.Show("請選擇交易所");
                return;
            }

            if (boxExchange.SelectedIndex == 0)
            {
                strExchangeNo = "US";
            }
            else if (boxExchange.SelectedIndex == 1)
            {
                strExchangeNo = "HK";
            }

            if (txtCancelSeqNo.Text.Trim() == "")
            {
                MessageBox.Show("請輸入欲刪除的委託序號");
                return;
            }
            strSeqNo = txtCancelSeqNo.Text.Trim();


            StringBuilder strMessage = new StringBuilder();
            strMessage.Length = 1024;
            Int32 nSiz = 1024;

            m_nCode = Functions.CancelForeignStockOrderBySeqNo(m_UserAccount, strSeqNo, strExchangeNo, strMessage, out nSiz);

            m_strMessage = strMessage.ToString();

            if (GetMessage != null)
            {
                GetMessage("CancelForeignStockOrder By SqlNo", m_nCode, m_strMessage);
            }
            strMessage = null;
        }

        #endregion

        #region Custom Method
        //----------------------------------------------------------------------
        // Custom Method
        //----------------------------------------------------------------------

        #endregion
    }
}

File: Capital/下單API_2.1.22/DotNet範例(VS2010)/OrderTesterVB/ForeignStockOrderControl.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Public Class ForeignStockOrderControl
#Region "Define Variable"
    '----------------------------------------------------------------------
    ' Define Variable
    '----------------------------------------------------------------------

    Private m_nCode As Integer
    Public m_strMessage As String

    Public Delegate Sub MyMessageHandler(ByVal strType As String, ByVal nCode As Integer, ByVal strMessage As String)
    Public Event GetMessage As MyMessageHandler

    Private m_UserAccount As String = ""
    Public Property UserAccount() As String
        Get
            Return m_UserAccount
        End Get
        Set(ByVal value As String)
            m_UserAccount = value
        End Set
    End Property

#End Region

    Private Sub btnSendForeignStockOrder_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSendForeignStockOrder.Click
        If m_UserAccount = "" Then
            MessageBox.Show("請選擇複委託帳號")
            Return
        End If

        Dim strExchangeNo As String = ""
        Dim strStockNo As String
        Dim nBidAsk As Integer
        Dim strPrice As String
        Dim nQty As Integer
        Dim strCurrency1 As String
        Dim strCurrency2 As String
        Dim strCurrency3 As String
        Dim nAccountType As Integer


        If boxAccountType.SelectedIndex < 0 Then
            MessageBox.Show("請選擇帳戶別")
            Return
        End If
        nAccountType = boxAccountType.SelectedIndex + 1

        If boxCurrency1.SelectedIndex < 0 Then
            MessageBox.Show("請至少選擇扣款幣別 1")
            Return
        End If
        strCurrency1 = boxCurrency1.Text
        strCurrency2 = boxCurrency2.Text
        strCurrency3 = boxCurrency3.Text

        If boxExchange.SelectedIndex < 0 Then
            MessageBox.Show("請選擇交易所")
            Return
        End If

        If boxExchange.SelectedIndex = 0 Then
            strExchangeNo = "US"
        ElseIf boxExchange.SelectedIndex = 1 Then
            strExchangeNo = "HK"
        End If

        If txtStockNo.Text.Trim() = "" Then
            MessageBox.Show("請輸入商品代碼")
            Return
        End If
        strStockNo = txtStockNo.Text.Trim()

        If boxBidAsk.SelectedIndex < 0 Then
            MessageBox.Show("請選擇買賣別")
            Return
        End If
        nBidAsk = boxBidAsk.SelectedIndex

        Dim dPrice As Double = 0.0
        If Double.TryParse(txtPrice.Text.Trim(), dPrice) = False Then
            MessageBox.Show("委託價請輸入數字")
            Return
        End If
        strPrice = txtPrice.Text.Trim()

        If Integer.TryParse(txtQty.Text.Trim(), nQty) = False Then
            MessageBox.Show("委託量請輸入數字")
            Return
        End If

        Dim strMessage As New StringBuilder()
        strMessage.Length = 1024
        Dim nSiz As Int32 = 1024

        m_nCode = Functions.SendForeignStockOrder(m_UserAccount, strStockNo, strExchangeNo, nBidAsk, strPrice, nQty, _
         strCurrency1, strCurrency2, strCurrency3, nAccountType, strMessage, nSiz)

        m_strMessage = strMessage.ToString()

        RaiseEvent GetMessage("ForeignStockOrder", m_nCode, m_strMessage)

        strMessage = Nothing

    End Sub

    Private Sub btnSendForeignStockOrderAsync_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSendForeignStockOrderAsync.Click
        If m_UserAccount = "" Then
            MessageBox.Show("請選擇複委託帳號")
            Return
        End If

        Dim strExchangeNo As String = ""
        Dim strStockNo As String
        Dim nBidAsk As Integer
        Dim strPrice As String
        Dim nQty As Integer
        Dim strCurrency1 As String
        Dim strCurrency2 As String
        Dim strCurrency3 As String
        Dim nAccountType As Integer


        If boxAccountType.SelectedIndex < 0 Then
            MessageBox.Show("請選擇帳戶別")
            Return
        End If
        nAccountType = boxAccountType.SelectedIndex + 1

        If boxCurrency1.SelectedIndex < 0 Then
            MessageBox.Show("請至少選擇扣款幣別 1")
            Return
        End If
        strCurrency1 = boxCurrency1.Text
        strCurrency2 = boxCurrency2.Text
        strCurrency3 = boxCurrency3.Text

        If boxExchange.SelectedIndex < 0 Then
            MessageBox.Show("請選擇交易所")
            Return
        End If

        If boxExchange.SelectedIndex = 0 Then
            strExchangeNo = "US"
        ElseIf boxExchange.SelectedIndex = 1 Then
            strExchangeNo = "HK"
        End If

        If txtStockNo.Text.Trim() = "" Then
            MessageBox.Show("請輸入商品代碼")
            Return
        End If
        strStockNo = txtStockNo.Text.Trim()

        If boxBidAsk.SelectedIndex < 0 Then
            MessageBox.Show("請選擇買賣別")
            Return
        End If
        nBidAsk = boxBidAsk.SelectedIndex

        Dim dPrice As Double = 0.0
        If Double.TryParse(txtPrice.Text.Trim(), dPrice) = False Then
            MessageBox.Show("委託價請輸入數字")
            Return
        End If
        strPrice = txtPrice.Text.Trim()

        If Integer.TryParse(txtQty.Text.Trim(), nQty) = False Then
            MessageBox.Show("委託量請輸入數字")
            Return
        End If

        m_nCode = Functions.SendForeignStockOrderAsync(m_UserAccount, strStockNo, strExchangeNo, nBidAsk, strPrice, nQty, _
         strCurrency1, strCurrency2, strCurrency3, nAccountType)

        RaiseEvent GetMessage("ForeignStockOrderAsync", m_nCode, "非同步委託")

    End Sub

    Private Sub btnCancelOrder_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancelOrder.Click
        Dim strExchangeNo As String = ""
        Dim strBookNo As String = ""

        If m_UserAccount = "" Then
            MessageBox.Show("請選擇複委託帳號")
            Return
        End If

        If boxExchange.SelectedIndex < 0 Then
            MessageBox.Show("請選擇交易所")
            Return
        End If

        If boxExchange.SelectedIndex = 0 Then
            strExchangeNo = "US"
        ElseIf boxExchange.SelectedIndex = 1 Then
            strExchangeNo = "HK"
        End If

        If txtCancelBookNo.Text.Trim() = "" Then
            MessageBox.Show("請輸入欲刪除的委託書號")
            Return
        End If
        strBookNo = txtCancelBookNo.Text.Trim()


        Dim strMessage As New StringBuilder()
        strMessage.Length = 1024
        Dim nSiz As Int32 = 1024

        m_nCode = Functions.CancelForeignStockOrderByBookNo(m_UserAccount, strBookNo, strExchangeNo, strMessage, nSiz)

        m_strMessage = strMessage.ToString()

        RaiseEvent GetMessage("CancelForeignStockOrder By BookNo", m_nCode, m_strMessage)

        strMessage = Nothing

    End Sub

    Private Sub btnCancelOrderBySeqNo_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancelOrderBySeqNo.Click
        Dim strExchangeNo As String = ""
        Dim strSeqNo As String = ""

        If m_UserAccount = "" Then
            MessageBox.Show("請選擇複委託帳號")
            Return
        End If

        If boxExchange.SelectedIndex < 0 Then
            MessageBox.Show("請選擇交易所")
            Return
        End If

        If boxExchange.SelectedIndex = 0 Then
            strExchangeNo = "US"
        ElseIf boxExchange.SelectedIndex = 1 Then
            strExchangeNo = "HK"
        End If

        If txtCancelSeqNo.Text.Trim() = "" Then
            MessageBox.Show("請輸入欲刪除的委託序號")
            Return
        End If
        strSeqNo = txtCancelSeqNo.Text.Trim()


        Dim strMessage As New StringBuilder()
        strMessage.Length = 1024
        Dim nSiz As Int32 = 1024

        m_nCode = Functions.CancelForeignStockOrderBySeqNo(m_UserAccount, strSeqNo, strExchangeNo, strMessage, nSiz)

        m_strMessage = strMessage.ToString()

        RaiseEvent GetMessage("CancelForeignStockOrder By SqlNo", m_nCode, m_strMessage)
        strMessage = Nothing

    End Sub
End Class
